### ISSUE TITLE
Add gradient clipping to Trainer to prevent training instability

### ISSUE THREAD

**Issue by luipin** [luipin/set-ml-module#26]
### Summary
The training configuration does not use gradient clipping. While OneCycleLR with AdamW is generally stable, gradient clipping (`gradient_clip_val`) is a low-cost safeguard against loss spikes — especially during the backbone fine-tuning phase (after `freeze_epochs`) when a large number of new parameters suddenly begin receiving gradient updates.

### Proposed change
Add `gradient_clip_val` when constructing the `pl.Trainer` in the notebook:

```python
trainer = pl.Trainer(
    max_epochs=30,
    gradient_clip_val=1.0,  # clip gradient norm to 1.0
    ...
)
```

A value of `1.0` is the standard default recommended in most PyTorch Lightning documentation and fine-tuning guides.

### Why the backbone unfreeze phase is risky
At epoch `freeze_epochs`, the backbone transitions from `requires_grad=False` to `requires_grad=True`. The optimizer's momentum/variance state for backbone parameters is zero-initialized at this point, so the first update step after unfreezing can produce large, unconditioned gradient steps. Gradient clipping limits the worst-case damage.

### Files affected
- `notebooks/set_card_classifier_main.ipynb`
